core/raftlease/fsm.go

```diff
@@ -101,35 +107,46 @@ func (f *FSM) extend(key lease.Key, holder string, duration time.Duration) *resp
 	return &response{}
 }
 
+func (f *FSM) pin(key lease.Key) *response {
+	f.pinned[key] = true
+	return &response{}
+}
+
+func (f *FSM) unpin(key lease.Key) *response {
+	delete(f.pinned, key)
+	return &response{}
+}
+
 func (f *FSM) setTime(oldTime, newTime time.Time) *response {
-	f.mu.Lock()
-	defer f.mu.Unlock()
 	if f.globalTime != oldTime {
 		return &response{err: globalclock.ErrConcurrentUpdate}
 	}
 	f.globalTime = newTime
+	return &response{expired: f.expired(newTime)}
```

Comment:
I don't think expired is a good name - it doesn't make it clear that it's deleting the leases at the same time. Maybe `removeExpired`?

Reply:
Done.